huaweicloud/services/workspace/data_source_huaweicloud_workspace_available_ip_number.go

```diff
@@ -0,0 +1,101 @@
+package workspace
+
+import (
+	"context"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+// @API Workspace GET /v2/{project_id}/subnets/{subnet_id}/available-ip
+func DataSourceAvailableIpNumber() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceAvailableIpNumberRead,
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: `The region in which the subnet is located.`,
+			},
+			"subnet_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `The ID of the subnet to be queried.`,
+			},
+			"available_ip": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: `The number of available IPs in the subnet.`,
+			},
+		},
+	}
+}
+
+func queryAvailableIpNumber(client *golangsdk.ServiceClient, d *schema.ResourceData) (interface{}, error) {
+	var (
+		httpUrl  = "v2/{project_id}/subnets/{subnet_id}/available-ip"
+		subnetId = d.Get("subnet_id").(string)
+	)
+
+	getPath := client.Endpoint + httpUrl
+	getPath = strings.ReplaceAll(getPath, "{project_id}", client.ProjectID)
+	getPath = strings.ReplaceAll(getPath, "{subnet_id}", subnetId)
+
+	opt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders: map[string]string{
+			"Content-Type": "application/json",
+		},
+	}
+
+	requestResp, err := client.Request("GET", getPath, &opt)
+	if err != nil {
+		return nil, err
+	}
+	respBody, err := utils.FlattenResponse(requestResp)
+	if err != nil {
+		return nil, err
+	}
+
+	availableIpNumber := utils.PathSearch("available_ip", respBody, "").(float64)
```

Comment:
do not set the different default type as your response value.
And the type of this value should be int (in the Windows System 32, the maximum is 2^31, but the maximum subnets just only 2^24 - 1)
```suggestion
	availableIpNumber := int(utils.PathSearch("available_ip", respBody, float64(0)).(float64))
```

Reply:
done